src/lib/database.ts

```diff
@@ -68,4 +68,36 @@ export class Database {
       completedMissions: updatedMissions,
     });
   }
+
+  /**
+   * Returns true if able to connect to the Firestore instance.
+   * The Firestore API times out a request after 60 seconds. This method
+   * implements a configurable override that defaults to 5 seconds, but there's
+   * no point in setting it higher than 60 seconds.
+   * @param timeout seconds
+   */
+  async isConnected(timeout: number = 5): Promise<boolean> {
+    try {
+      timeout = Math.min(timeout, 60) * 1000;
+      // eslint-disable-next-line no-undef
+      let timerId: NodeJS.Timeout;
+
+      const timer = new Promise<boolean>((resolve) => {
+        timerId = setTimeout(() => resolve(false), timeout);
+      });
+
+      // TODO: research if there's a lighter weight way to status a connection.
+      const connectionCheck = this.db.listCollections();
+
+      return Promise.race([connectionCheck, timer]).then(result => {
+        clearTimeout(timerId);
+        return !!result;
+      });
+
+    } catch (err) {
+      // GoogleError: Total timeout of API google.firestore.v1.Firestore
```

Comment:
Not a fan of silent errors. Anyway to capture the errors rather than just seeing `isConnected` returned `false`.

Reply:
Yeah, I thought about that. Bottom line is it's not healthy yet. If not the expected error, it could be logged if this turns out to be problematic.

Reply:
Mind if I file an issue for this as well to do a revisit?